tests/components/broadlink/test_sensors.py

```diff
@@ -0,0 +1,287 @@
+    mock_entry = device.get_mock_entry()
+    mock_entry.add_to_hass(hass)
+
+    device_registry = mock_device_registry(hass)
+    entity_registry = mock_registry(hass)
+
+    with patch("broadlink.gendevice", return_value=mock_api):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    device_entry = device_registry.async_get_device(
+        {(DOMAIN, mock_entry.unique_id)}, set()
+    )
+    entries = async_entries_for_device(entity_registry, device_entry.id)
+    sensors = {entry for entry in entries if entry.domain == SENSOR_DOMAIN}
+    assert len(sensors) == 1
+
+    mock_api.check_sensors.return_value = {"temperature": 25.8}
+    await hass.helpers.entity_component.async_update_entity(
+        next(iter(sensors)).entity_id
+    )
+    assert mock_api.check_sensors.call_count == 2
+
+    sensors_and_states = {
+        (sensor.original_name, hass.states.get(sensor.entity_id).state)
+        for sensor in sensors
+    }
+    assert sensors_and_states == {(f"{device.name} Temperature", "25.8")}
+
+
+async def test_rm_mini3_no_sensor(hass):
+    """Test we do not set up sensors for RM mini 3."""
+    device = get_device("Entrance")
+    mock_api = device.get_mock_api()
+    mock_api.check_sensors.return_value = {"temperature": 0}
+    mock_entry = device.get_mock_entry()
+    mock_entry.add_to_hass(hass)
+
+    device_registry = mock_device_registry(hass)
+    entity_registry = mock_registry(hass)
+
+    with patch("broadlink.gendevice", return_value=mock_api):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert mock_api.check_sensors.call_count <= 1
+    device_entry = device_registry.async_get_device(
+        {(DOMAIN, mock_entry.unique_id)}, set()
+    )
+    entries = async_entries_for_device(entity_registry, device_entry.id)
+    sensors = {entry for entry in entries if entry.domain == SENSOR_DOMAIN}
+    assert len(sensors) == 0
+
+
+async def test_rm4_pro_hts2_sensor_setup(hass):
+    """Test a successful RM4 pro sensor setup with HTS2 cable."""
+    device = get_device("Garage")
+    mock_api = device.get_mock_api()
+    mock_api.check_sensors.return_value = {"temperature": 22.5, "humidity": 43.7}
+    mock_entry = device.get_mock_entry()
+    mock_entry.add_to_hass(hass)
+
+    device_registry = mock_device_registry(hass)
+    entity_registry = mock_registry(hass)
+
+    with patch("broadlink.gendevice", return_value=mock_api):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert mock_api.check_sensors.call_count == 1
+    device_entry = device_registry.async_get_device(
+        {(DOMAIN, mock_entry.unique_id)}, set()
+    )
+    entries = async_entries_for_device(entity_registry, device_entry.id)
+    sensors = {entry for entry in entries if entry.domain == SENSOR_DOMAIN}
+    assert len(sensors) == 2
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6072450)